lib/_stream_readable.js

```diff
@@ -851,6 +851,22 @@ Readable.prototype.wrap = function(stream) {
   return self;
 };
 
+Object.defineProperty(Readable.prototype, 'readBuffer', {
+  get: function() {
+    return this._readableState.buffer;
+  }
+});
+
+Object.defineProperty(Readable.prototype, 'flowing', {
+  get: function() {
+    return this._readableState.flowing;
+  }
+});
+
+// so that http_client can set it
+Readable.prototype._setFlowing = function(state) {
+  this._readableState.flowing = state;
```

Comment:
Same for all those three. In the constructor `this._readableState` might not be populated.